### PR TITLE
[ Feature request] : addon of jule program for reverse the string

### PR DESCRIPTION
**Pull request by Jivanjamadar**
added 2 files of program based on jule programming language:
1) StringReverse.jule : contains the main logic for reversing the string
2) Stringreversetest.jule: contains necessary test cases for testing this code in jule

### CODE REVIEW

**Comment by Jivanjamadar** [comment]
@mertcandav please review this

File: strings/StringReverseTest.jule [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    },
+    
+    ]
+    
+    #test
+    fn testStringRevese(t: &testing::T){
+        for_, test in StringReverseTests{
+            funcResult := StringReverse(test.s)
+            if test.expected != funcResult{
+                t.Error("Expected: {}, got{}", test.expected, funcResult)
```
Comment (mertcandav) [inline strings/StringReverseTest.jule:69]: `testing::T` is not have the `Error` method.
It should be `Errorf`.
```suggestion
                t.Errorf("Expected: {}, got{}", test.expected, funcResult)
```

File: strings/StringReverseTest.jule [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        "String with spaces only",
+        "   ",
+        "   ",
+    },
+    
+    ]
+    
+    #test
+    fn testStringRevese(t: &testing::T){
+        for_, test in StringReverseTests{
```
Comment (mertcandav) [inline strings/StringReverseTest.jule:66]: The `for_` is a invalid syntax. We need to separate it.
```suggestion
        for _, test in StringReverseTests {
```

File: strings/StringReverseTest.jule [hunk 3]
```diff
@@ -0,0 +1,72 @@
+#build test
+
+use "std/testing"
+struc StringReverseTest {
```
Comment (mertcandav) [inline strings/StringReverseTest.jule:4]: There is no `struc` keyword, it should be `struct`.
```suggestion
struct StringReverseTest {
```

File: strings/StringReverse.jule
Comment (mertcandav) [inline strings/StringReverse.jule:1]: This is a totally a wrong and inefficient implementation.
Jule have unicode support, so unicode characters should be handled correctly.
You are reversing string byte-by-byte. You have to reverse rune-by-rune.

File: strings/StringReverse.jule [hunk 4]
```diff
@@ -0,0 +1,12 @@
+// Reverse a given string and return reversed string 
+// e.g. input: Hello, output: olleH
+// Author: Jivanjamadar
+
+fn StringReverse(s: str): str{
```
Comment (mertcandav) [inline strings/StringReverse.jule:5]: This function defined in the `strings` package.
So, naming can be more simple.
```suggestion
fn Reverse(s: str): str {
```

**Review by mertcandav — CHANGES_REQUESTED** [review]

**Review by mertcandav — COMMENTED** [review]
Please rename files:
- `StringReverse.jule` as `reverse.jule`
- `StringReverseTest.jule` as `reverse_test.jule`

File: strings/StringReverse.jule
Comment (Jivanjamadar) [reply strings/StringReverse.jule:1]: Extremely sorry sir, Jule is very new language for me and for most of developers. I getting wrong in its syntax. I am trying my best to do that. Please, give me some time. I will try to correct it and get back to you. Thank you

Comment (mertcandav) [reply strings/StringReverse.jule:1]: It's ok. You can request a review whenever you ready.
Thank you for your willingness to contribute.

**Comment by Jivanjamadar** [comment]
@mertcandav I have done changes which you requested. Please review the program files of test & main logic and give your reveiw for futrue changes, Thank you!

File: strings/reverse.jule [hunk 5]
```diff
@@ -0,0 +1,15 @@
+// Reverse a given string and return reversed string 
+// e.g. input: Hello, output: olleH
+// Author: Jivanjamadar
+
+fn StringReverse(s: str): str {
+    runes := s.runes()                // Convert string to rune array
+    mut reversed := []rune(len(runes)) // Create an array of the same length as 'runes'
+
+
+    for i := 0; i < len(runes); i++ {
```
Comment (mertcandav) [inline strings/reverse.jule:10]: Invalid iteration syntax.
```suggestion
    mut i := 0
    for i < len(runes); i++ {
```

**Review by mertcandav — CHANGES_REQUESTED** [review]
There are still a few issues other than the ones I mentioned. Please fix the implementation issues described, then I will merge this PR and fix the remaining issues.

**Review by mertcandav — APPROVED** [review]
LGTM. Thanks for your contributions.

**Pull request merged**